### PR TITLE
Add type hints and update tests/docstrings in mathematics

### PR DESCRIPTION
**Pull request by namurphy**

### CODE REVIEW

File: plasmapy/mathematics/mathematics.py [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 from mpmath import polylog
 from scipy.special import wofz as Faddeeva_function
+from typing import Union
 
-# TODO: Add type hint annotations to this module.
 
-
-def plasma_dispersion_func(zeta):
+def plasma_dispersion_func(zeta: Union[complex, int, float, np.ndarray, u.Quantity]
+                           ) -> Union[complex, float, u.Quantity]:
```
Comment (StanczakDominik) [inline plasmapy/mathematics/mathematics.py]: I think we're missing `np.ndarray` on the return end here. Seems to me like if you pass an array in, an array should come out.

File: plasmapy/mathematics/mathematics.py [hunk 2]
```diff
@@ -92,7 +91,8 @@ def plasma_dispersion_func(zeta):
     return Z
 
 
-def plasma_dispersion_func_deriv(zeta):
+def plasma_dispersion_func_deriv(zeta: Union[complex, int, float, np.ndarray, u.Quantity]
+                                 ) -> Union[complex, float, u.Quantity]:
```
Comment (StanczakDominik) [inline plasmapy/mathematics/mathematics.py]: Likewise here.

File: plasmapy/mathematics/mathematics.py [hunk 3]
```diff
@@ -194,32 +194,30 @@ def Fermi_integral(x, j):
...
-    See also
+    See Also
     --------
     plasma_dispersion_func
 
     Notes
     -----
-    The complete Fermi-Dirac integral is defined as: [1]_
+    The `complete Fermi-Dirac integral
+    <https://en.wikipedia.org/wiki/Complete_Fermi-Dirac_integral>`_ is
```
Comment (StanczakDominik) [inline plasmapy/mathematics/mathematics.py]: Why the change in links here?

Comment (namurphy) [reply plasmapy/mathematics/mathematics.py]: I was hoping to have this appear as an in-line link, which makes it easier to get to compared to the references.

File: plasmapy/mathematics/mathematics.py (hunk 1 above)
Comment (namurphy) [reply plasmapy/mathematics/mathematics.py]: Good catch!  It's turning out to be pretty easy to miss type hints...
